File: src/ros2_bridge/mappers/unique_identifier_msgs/uuid.rs
//! Typed mapper for `unique_identifier_msgs/msg/UUID`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn uuid_to_bus(msg: ros_env::unique_identifier_msgs::msg::UUID) -> crate::unique_identifier_msgs::msg::v1::Uuid {
    crate::unique_identifier_msgs::msg::v1::Uuid {
        uuid: crate::ros2_bridge::mappers::convert::IntoU8Vec::into_u8_vec(msg.uuid),
    }
}

pub(crate) fn uuid_to_ros(bus: crate::unique_identifier_msgs::msg::v1::Uuid) -> ros_env::unique_identifier_msgs::msg::UUID {
    ros_env::unique_identifier_msgs::msg::UUID {
        uuid: crate::ros2_bridge::mappers::convert::FromByteSeq::from_byte_seq(bus.uuid),
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct UniqueIdentifierMsgsUuidMapper;

impl TypedTopicMapper for UniqueIdentifierMsgsUuidMapper {
    type Ros = ros_env::unique_identifier_msgs::msg::UUID;
    type Bus = crate::unique_identifier_msgs::msg::v1::Uuid;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(uuid_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(uuid_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/unique_identifier_msgs/uuid.py
"""Generated mapper for `unique_identifier_msgs/msg/UUID`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.unique_identifier_msgs.msg.v1 import UUID as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def uuid_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.uuid = bytes(msg.uuid)
    return bus


def uuid_to_ros(bus):
    from unique_identifier_msgs.msg import UUID as RosMsg

    out = RosMsg()
    out.uuid = bytes(bus.uuid)
    return out


class UniqueIdentifierMsgsUuidMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from unique_identifier_msgs.msg import UUID as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return uuid_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return uuid_to_ros(bus)
